### DISCUSSION TITLE
Upload blocks from ingester are uploaded to backend periodically i.e. after 2 hrs by default. What is parameter to change this time in Mimir

### DISCUSSION THREAD

**Discussion by Shindek77** [grafana/mimir#5072] [Help and support]
As per grafana mimir documnetation :
Incoming series from [distributors] are not immediately written to the long-term storage but are either kept in ingesters memory or offloaded to ingesters disk. Eventually, all series are written to disk and periodically uploaded (**by default every two hours)** to the long-term storage.

But I want to change this default time is there any parameter in config file using that I can do this. Please help

**Comment by charleskorn** [comment]
Why would you like to change this? We generally don't recommend changing the upload interval.

**Comment by Shindek77** [comment]
Actually it our project requirement where we want to change default 2 hours time. But for changing this I am not able to find out which parameter in configuration file i.e. values.yaml of mimir helm chart needs to be changed.

anyone know the parameter name please let me know?

**Reply by pstibrany** [reply]
Simple version is that data is uploaded in blocks and blocks are created every `-blocks-storage.tsdb.block-ranges-period`. It defaults to 2h. As Charles wrote, we do not recommend changing it... If anything breaks, you're on your own.

**Reply by Shindek77** [reply]
@pstibrany  ok thank you for your input..Could you pleas help me what is exact difference between blocks-storage.tsdb.head_compaction_interval and -blocks-storage.tsdb.block-ranges-period parameters. I am bit confused about them

**Reply by pstibrany** [reply]
`head_compaction_interval` specifies how often Mimir calls "Compact()" method on TSDB Head. This call corresponds to asking "TSDB Head, would you like to run compaction now?". TSDB Head then decides whether compaction needs to run or not -- based on how much data it has in memory. Specifically it requires `3/2 * block range` of data in memory. For default value of "block range" -- 2h, it needs 3h of data in memory. If this condition doesn't hold, TSDB Head will ignore compaction request. If condition holds, and compaction runs, Head stores first 2h (block range) of data into block and removes it from memory, keeping remaining 1h in memory.

`-blocks-storage.tsdb.block-ranges-period` controls this block range.
